manifests/backup/pg_dump.pp

```diff
@@ -0,0 +1,151 @@
+  Boolean $delete_before_dump = false,
+  String $dir,
+  String $dir_group = '0',
+  String $dir_mode = '0700',
+  String $dir_owner = 'root',
+  Enum['present','absent'] $ensure = 'present',
+  Enum['plain','custom','directory','tar'] $format = 'plain',
+  Boolean $install_cron = true,
+  Boolean $manage_user = false,
+  Array $optional_args = [],
+  String $pgpass_path = '/root/.pgpass',
+  Integer $rotate = 30,
+  String $script_path = '/usr/local/sbin/pg_dump.sh',
+  String $success_file_path = '/tmp/pgbackup_success',
+  String $template = 'postgresql/pg_dump.sh.epp',
+  Array $time = ['23', '5'],
+  String $weekday = '*',
+  Optional[Variant[String, Sensitive[String]]] $db_password = undef,
+  Optional[String] $db_user = undef,
+  Optional[String] $package_name = undef,
+  Optional[String] $post_script = undef,
+  Optional[String] $pre_script = undef,
+) {
+  # Install required packages
+  if $package_name {
+    ensure_packages($package_name)
+  }
+  if $install_cron {
+    if $facts['os']['family'] == 'RedHat' {
+      ensure_packages('cronie')
+    } elsif $facts['os']['family'] != 'FreeBSD' {
+      ensure_packages('cron')
+    }
+  }
+
+  # Setup db user with required permissions
+  if $manage_user and $db_user and $db_password {
+    # Create user with superuser privileges
+    postgresql::server::role { $db_user:
+      ensure        => $ensure,
+      password_hash => postgresql::postgresql_password($db_user, $db_password),
+      superuser     => true,
+    }
+
+    # Allow authentication from localhost
+    postgresql::server::pg_hba_rule { 'local access as backup user':
+      type        => 'local',
+      database    => 'all',
+      user        => $db_user,
+      auth_method => 'md5',
+      order       => 1,
+    }
+  }
+
+  # Create backup directory
+  file { $dir:
+    ensure => 'directory',
+    mode   => $dir_mode,
+    owner  => $dir_owner,
+    group  => $dir_group,
+  }
+
+  # Create backup script
+  file { $script_path:
+    ensure  => $ensure,
+    mode    => '0700',
+    owner   => 'root',
+    group   => '0', # Use GID for compat with Linux and BSD.
+    content => epp($template, {
+      compress => $compress,
+      databases => $databases,
+      db_user => $db_user,
+      delete_before_dump => $delete_before_dump,
+      dir => $dir,
+      format => $format,
+      optional_args => $optional_args,
+      post_script => $post_script,
+      pre_script => $pre_script,
+      rotate => $rotate,
+      success_file_path => $success_file_path,
+    }),
+  }
+
+  # Create password file for pg_dump
+  file { $pgpass_path:
+    ensure    => $ensure,
+    mode      => '0600',
+    owner     => 'root',
+    group     => '0', # Use GID for compat with Linux and BSD.
+    content   => inline_epp('*:*:*:<%= $db_user %>:<%= $db_password %>',{
+      db_password => $db_password,
+      db_user => $db_user,
+    }),
+    show_diff => false,
+  }
+
+  # Create cron job
+  cron { 'pg_dump backup job':
```

Comment:
did you consider a systemd timer? crons are quite old and systemd is available on every linux platform this modules supports. IMO that should be used.

Reply:
I use cron on purpose for compatibility reasons. I also plan to add (official) support for FreeBSD, which is my target operating system for this feature/module. :)

Reply:
I still consider crons legacy and I don't think we should keep deploying them on linux. maybe the Puppet teams sees that different.

Reply:
Sure, opinions and experiences may differ. Cron plays a vital role in all environments that *I* care for, and it's not going anywhere. So in *my* experience cron is neither legacy nor outdated. Besides that, I think it is desirable to use the same tool on as many platforms as possible, it reduces complexity.